### PR TITLE
[WX-1833] Use monaco-editor in new WDLEditor

### PR DESCRIPTION
**Pull request by kpierre13**
### What
- Adding the monaco-editor to Terra-UI for use as a code editor

### Why
- The Firecloud UI to Terra UI migration requires the ability for users to paste/write/edit their WDLs. To do this, we would need a viable code-editor, however, it's the existing WDLViewer is read-only and does not allow editing. So, before completely replacing the WDLViewer, we've added the monaco-editor which may be well known to VSCode users.

### Testing strategy

- [x] Manual testing
- [x] Unit testing

Pre-existing WDL from the Workflows tab in GCP:
image

Addition of small window in top right-hand side of box:
image

Ability to make the WDLViewer read-only:
image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,6 +8,7 @@
     "@dnd-kit/sortable": "^7.0.2",
     "@dnd-kit/utilities": "^3.2.1",
     "@microlink/react-json-view": "^1.22.2",
+    "@monaco-editor/react": "^4.6.0",
```
Comment (cahrens) [inline package.json]: Please also remove `prismjs` as there don't appear to be any other usages of it.

Comment (kpierre13) [reply package.json]: Update: it seems like the monaco-editor/react package is pretty challenging to unit test, and I felt uncomfortable replacing all instances of `WDLViewer` with this package only to have pre-existing unit tests break without a clean way to fix them. The rest of the team also felt very strongly about using this editor, so in an attempt to not scrap it completely we've opted for the current solution:

- Keep the existing `WDLViewer`. Any updates to its uses have been reverted.
- import `monaco-editor/react` package and use in a new `WDLEditor` component.
- For now, the `WDLEditor` will be contained within the `workflows` folder and only used by one component, `WorkflowModal`.
- Although I'm unable at this time (for the sake of the migration) to get the `WDLEditor` to render any wdl content, I did create a unit test that checks for the styling on the Editor element to verify that it does load onto the page.

File: src/pages/workflows/common/WDLEditor.tsx [hunk 2]
```diff
@@ -0,0 +1,178 @@
+import Editor from '@monaco-editor/react';
+import React from 'react';
+
+/* The syntax highlighting is not exactly what is used to be with the addition
+ * of the new editor. This ticket is a follow-up to update the syntax highlighting
+ * to better resemble what we had using PrismJs: https://broadworkbench.atlassian.net/browse/WX-1838
+ */
```
Comment (aednichols) [inline src/pages/workflows/common/WDLEditor.tsx:7]: Another thing that I _think_ is true: Monaco can use syntax highlighting from VS Code, and there are community-created highlighters for WDL

**Review by aednichols — COMMENTED** [review]
I am extremely happy that we found a middle path to start adopting Monaco.

While testing on the PR environment, the editor hung on "Loading..." and issued a `Monaco initialization: error` in the console. I didn't dig deeper into the error, it seemed frustratingly nonspecific.

Screenshot 2024-09-06 at 16 04 05

File: src/pages/workflows/common/WDLEditor.tsx (hunk 2 above)
Comment (aednichols) [reply src/pages/workflows/common/WDLEditor.tsx:7]: I should clarify – maybe add that fact to the comment, don't necessarily need to implement it immediately.

**Comment by kpierre13** [comment]
> I am extremely happy that we found a middle path to start adopting Monaco.
> 
> While testing on the PR environment, the editor hung on "Loading..." and issued a `Monaco initialization: error` in the console. I didn't dig deeper into the error, it seemed frustratingly nonspecific.

Hm, it seems like the deployment mocks a production environment, and the CSP is blocking `https://cdn.jsdelivr.net/npm/monaco-editor@0.43.0/min/vs/loader.js`... I'm not too familiar with our policies regarding adding/altering the allowed sources. I'll draft up a message and post it in the Terra UI channel on Monday. Thanks for catching! @aednichols

**Comment by aednichols** [comment]
Seems like a reasonable strategy. Looks like we add added the CDNs for AppCues and Cloudflare already, so an incremental Javascript CDN seems appropriate.

https://github.com/DataBiosphere/terra-ui/blob/31e3bf295d8d754edf2b4fac1016658f24ed5ae8/app.yaml#L27

**Review by aednichols — APPROVED** [review]
LGTM 👍 

Tested on PR environment, I can paste in and edit WDL. I get a "not yet implemented" message on save, which seems completely acceptable at this stage.

**Review by pshapiro4broad — APPROVED** [review]
Looks good to me. Thanks for picking this work up!

**Pull request merged**